WordPress/gutenberg
# Style Engine: add typography to backend

Tracking Issue: https://github.com/WordPress/gutenberg/issues/38167

## What?

Following on from https://github.com/WordPress/gutenberg/pull/39446 that introduced spacing to the Style Engine backend, this PR takes on typography ⚔️ 

## Why?

To introduce the Style Engine incrementally at first, and not change any output for now.

## How?

1. Using an existing method to output inline styles `generate`
2. Creating a new method `get_classnames` to output classnames inc. slugs.

## Testing Instructions

For dynamic blocks (Site Title, Navigation, Post Title, Post Date for example), check that the typography is output correctly on the frontend.

I've been testing so far with the Site Title block. To test with all controls you can update that [block's JSON](https://github.com/WordPress/gutenberg/blob/eaa2e0a91a8ccac43fa2e39f60c37713bff165d7/packages/block-library/src/site-title/block.json) supports with the following:

```json
		"typography": {
			"fontSize": true,
			"lineHeight": true,
			"__experimentalFontFamily": true,
			"__experimentalTextTransform": true,
			"__experimentalFontStyle": true,
			"__experimentalFontWeight": true,
			"__experimentalLetterSpacing": true,
			"__experimentalTextDecoration": true,
			"__experimentalDefaultControls": {
				"fontSize": true,
				"lineHeight": true,
				"fontAppearance": true,
				"letterSpacing": true,
				"textTransform": true
			}
		}
```

Test:

- Preset color classnames and font sizez
- Inline styles

Some test code (using TwentyTwentyTwo theme)

```html

```

Deprecated HTML with presets
```html



```
## Screenshots or screencast
